core/upgrade/src/main/java/org/opennms/upgrade/implementations/EventParameterMigratorOffline.java

```diff
@@ -0,0 +1,147 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2017 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2017 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.upgrade.implementations;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Map;
+
+import org.opennms.core.db.DataSourceFactory;
+import org.opennms.netmgt.events.api.EventParameterUtils;
+import org.opennms.netmgt.xml.event.Parm;
+import org.opennms.upgrade.api.AbstractOnmsUpgrade;
+import org.opennms.upgrade.api.OnmsUpgradeException;
+
+public class EventParameterMigratorOffline extends AbstractOnmsUpgrade {
+
+    public EventParameterMigratorOffline() throws OnmsUpgradeException {
+        super();
+    }
+
+    @Override
+    public int getOrder() {
+        return 8;
+    }
+
+    @Override
+    public String getDescription() {
+        return "Moves event parameters from eventparms column to table";
+    }
+
+    @Override
+    public boolean requiresOnmsRunning() {
+        return false;
+    }
+
+    @Override
+    public void preExecute() throws OnmsUpgradeException {
+        ;
+        // check for eventparms column
```

Comment:
Is this still necessary given that we run the `preMigrationStatement` in the `execute` call?